pkg/healthcheck/healthcheck.go

```diff
@@ -2264,34 +2258,154 @@ func (hc *HealthChecker) checkMisconfiguredOpaquePortAnnotations(ctx context.Con
 	return nil
 }
 
-func misconfiguredOpaquePortAnnotationsInService(service *corev1.Service, pods []*corev1.Pod) error {
-	for _, pod := range pods {
-		if err := misconfiguredOpaqueAnnotation(service, pod); err != nil {
-			return err
+// getEndpointsPods takes a collection of endpoints and returns a map of all
```

Comment:
TIOLI: I might change this comment to say that it returns a "set" instead of a "map" since we're using maps as sets here.